### PR TITLE
Make each rsnotify listener implementation a module

### PR DESCRIPTION
**Pull request by jonyoder**
* Makes each `Listener` implementation a module of its own. This will make it easy to import only the listener types you need.
* Moves all of the listener factory implementations into the correct `listeners` implementation packages.
* Refactors some naming per Go recommendations. For example "local.LocalListener" becomes "local.Listener".

### CODE REVIEW

File: pkg/rsnotify/listenerfactory/listenerfactory.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package listenerfactory
```
Comment (jonyoder) [inline pkg/rsnotify/listenerfactory/listenerfactory.go:1]: The code here was moved from `factory/listenerfactory.go`. None of it has changed, apart from removing the implementations and moving them to the correct listener implementation packages.

File: pkg/rsnotify/listeners/local/factory.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package local
```
Comment (jonyoder) [inline pkg/rsnotify/listeners/local/factory.go:1]: Moved from the old factory. This code remains basically unchanged.

File: pkg/rsnotify/listeners/local/locallistener.go [hunk 3]
```diff
@@ -13,34 +13,34 @@ import (
...
+	guid     string
+	provider *ListenerProvider
 
 	// If this channel is non-nil, listening is deferred until this channel
 	// is signaled or closed. Currently used in test only.
 	deferredStart chan struct{}
 }
 
-type LocalListenerFactory struct {
+type ListenerProvider struct {
```
Comment (jonyoder) [inline pkg/rsnotify/listeners/local/locallistener.go:28]: This name conflicted with the factory and `Provider` seemed appropriate.

File: pkg/rsnotify/listeners/postgrespgx/factory.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
+package postgrespgx
```
Comment (jonyoder) [inline pkg/rsnotify/listeners/postgrespgx/factory.go:1]: Moved from the old factory. This code remains basically unchanged.

File: pkg/rsnotify/listeners/postgrespq/factory.go [hunk 5]
```diff
@@ -0,0 +1,41 @@
+package postgrespq
```
Comment (jonyoder) [inline pkg/rsnotify/listeners/postgrespq/factory.go:1]: Moved from the old factory. This code remains basically unchanged.

**Review by tylfin — APPROVED** [review]
This all makes sense for me.

For projects that require both, could be dependency requirements that demand they be upgraded in step anyways?

**Pull request merged**